File: Objective-C/macOS SDK v4 Sample/ActivateViewController.m
//
//  ActivateViewController.m
//  macOS SDK v4 Sample
//
//  Created by Louis Harwood on 25/11/2020.
//  Copyright © 2020 Paddle.com LTD. All rights reserved.
//

#import "ActivateViewController.h"
#import "AppConfig.h"
@import Paddle;

@interface ActivateViewController () <PADProductDelegate>

@end

@implementation ActivateViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do view setup here.
}

- (IBAction)cancelButton:(id)sender {
    [self.view.window close];
}

- (IBAction)nextButton:(id)sender {
    if (self.productSelection.indexOfSelectedItem == 0) {
        [Paddle.sharedInstance showLicenseActivationDialogForProduct:[self mainProduct] email:nil licenseCode:nil activationStatusCompletion:^(PADActivationState activationState) {
            if (activationState == PADActivationActivated) {
                [self.view.window close];
                [[NSNotificationCenter defaultCenter] postNotificationName:@"activationsChanged" object:self];
            }
        }];
    } else {
        [Paddle.sharedInstance showLicenseActivationDialogForProduct:[self secondProduct] email:nil licenseCode:nil activationStatusCompletion:^(PADActivationState activationState) {
            if (activationState == PADActivationActivated) {
                [self.view.window close];
                [[NSNotificationCenter defaultCenter] postNotificationName:@"activationsChanged" object:self];
            }
        }];
    }
}



#pragma mark - Paddle helper methods

- (PADProduct *)mainProduct
{
    PADProduct *product = [[PADProduct alloc] initWithProductID:PAD_SDK_PRODUCT_ID
                                                    productType:PADProductTypeSDKProduct
                                                  configuration:[AppConfig configurationForMainProduct]];
    product.delegate = self;
    product.localizedTrialText = NSLocalizedStringFromTableInBundle(@"Thank you for trying out the Random Number Generator 500", nil, [NSBundle mainBundle], nil);
    return product;
}

- (PADProduct *)secondProduct
{
    PADProduct *product = [[PADProduct alloc] initWithProductID:PAD_SDK_SECOND_PRODUCT_ID
                                                    productType:PADProductTypeSDKProduct
                                                  configuration:[AppConfig configurationForSecondProduct]];
    product.delegate = self;

    return product;
}

@end

File: Swift/macOS SDK v4 Swift Sample/ActivateViewController.swift
//
//  ActivateViewController.swift
//  macOS SDK v4 Swift Sample
//
//  Created by Louis Harwood on 27/05/2021.
//  Copyright © 2021 Paddle. All rights reserved.
//

import Foundation
import Cocoa
import Paddle

class ActivateViewController: NSViewController, PADProductDelegate {
    @IBOutlet var productSelection: NSPopUpButton!
    
    @IBAction func cancelButton(sender: NSButton) {
        view.window?.close()
    }
    
    @IBAction func nextButton(sender: NSButton) {
        if productSelection.indexOfSelectedItem == 0 {
            Paddle.sharedInstance()?.showLicenseActivationDialog(for: mainProduct(), email: nil, licenseCode: nil, activationStatusCompletion: { activationState in
                if activationState == PADActivationState.activated {
                    self.view.window?.close()
                }
            })
        } else {
            Paddle.sharedInstance()?.showLicenseActivationDialog(for: secondProduct(), email: nil, licenseCode: nil, activationStatusCompletion: { activationState in
                if activationState == PADActivationState.activated {
                    self.view.window?.close()
                }
            })
        }
    }
    
    func mainProduct() -> PADProduct {
        let product = PADProduct.init(productID: AppConfig.PAD_SDK_PRODUCT_ID, productType: PADProductType.sdkProduct, configuration: AppConfig.configurationForMainProduct())
        product?.delegate = self
        
        return product!
    }
    
    func secondProduct() -> PADProduct {
        let product = PADProduct.init(productID: AppConfig.PAD_SDK_SECOND_PRODUCT_ID, productType: PADProductType.sdkProduct, configuration: AppConfig.configurationForSecondProduct())
        product?.delegate = self
        
        return product!
    }
}
